File: algorithm/leetcode/classical150item/interval/findMinArrowShots.go
package main

import (
	"sort"
)

func findMinArrowShots(points [][]int) int {
	sort.Slice(points, func(i, j int) bool {
		return points[i][1] < points[j][1]
	})
	right := points[0][1]
	ans := 1
	for _, v := range points {
		if v[0] > right {
			right = v[1]
			ans++
		}
	}
	return ans
}

File: algorithm/leetcode/classical150item/interval/findMinArrowShots.py
from typing import List


class Solution:
    def findMinArrowShots(self, points: List[List[int]]) -> int:
        if not points:
            return 0
        points.sort(key=lambda p: p[1])
        right = points[0][1]
        ans = 1
        for p in points:
            if p[0] > right:
                ans += 1
                right = p[1]
        return ans
